File: state/kotlin/InitializeState.kt
package State

// Farklı durumların meydana geldiği sınıf.
// UML diyagramındaki State (VendingMachineState) yapısından türer.
// UML diyagramındaki ConcreteState yapısına denk gelmektedir.
internal class InitializeState : VendingMachineState() {
    // Başlangıç durumu ayarları yapıldıktan sonra WaitingState (hazır) durumuna geçer.
    override fun handleState(context: VendingMachine) {
        println("InitializeState: Ön hazırlıklar yapılıyor.")

        context.setState(WaitingState())
    }
}

File: state/python/initialize_state.py
from vending_machine_state import VendingMachineState
from waiting_state import WaitingState


class InitializeState(VendingMachineState):
    """
    Farklı durumların meydana geldiği sınıf.
    UML diyagramındaki State (VendingMachineState) yapısından türer.
    UML diyagramındaki ConcreteState yapısına denk gelmektedir.
    """

    def handle_state(self, context):
        # Başlangıç durumu ayarları yapıldıktan sonra WaitingState (hazır) durumuna geçer.
        print("InitializeState: Ön hazırlıklar yapılıyor.")
        context.set_state(WaitingState())
